Directory overhaul to enhance clarity for users

spaulaus opened the pull request:
I  overhauled the directory structure to make things clearer for the users. We now group the Acquisition (poll, MCA, etc.) into a unified folder as it has been with Scan (renamed to Analysis). This necessitated a number of changes to the CMakeLists.txt files for these directories.  

I  renamed some of the static and shared libraries so that they are a little bit clearer. For example, PixieCoreStatic is now PaassCoreStatic. I think it makes things a little clearer since we've got other things named "Core" floating around. 

I created the Resource and ThirdParty directories. These directories house software that are useful to many different pieces of the software. Previously, the software contained within these directories was contained solely in the Scan (now Analysis) directory. This will allow fixes for things like #72 to be more useful, since the code necessary to fix this can be applied to unpacking the data as well. 

The ThirdParty software compiled with the code is only pugixml. However, if we move to the Google Testing framework then we would be able to add this software here. This also makes it possible to include some dedicated minimization software that doesn't rely on GSL or ROOT. 

P.S. We need a label for Affects literally everything.

spaulaus committed:
Massive overhaul to directory structure. 

We needed to overhaul the directory structure so that we could start implementing changes that could be seen by all of the software and not jus a subset. In addition, the analysis software (Scan) all lived in its own directory, where as the Acquisition (Poll, MCA, etc) all lived in the base directory. Simplification of the structure means that the users will have an easier time finding the information that they need.

spaulaus committed:
Adjusting the CMakeLists files for new compilation paths

I have also adjusted the name of the different libraries so that they are more explicit. For example, there was PixieCoreStatic, which was renamed to PaassCoreStatic. This makes things slightly more focused on the software that we're building and not on Pixie in general (ignore the P in PAASS). The pugixml software has been moved to ThirdParty.

spaulaus committed:
Populating the Resources directory with things that will be useful.

ksmith0 commented on CMakeLists.txt:
```diff
@@ -65,17 +69,16 @@ option(USE_ROOT "Use ROOT" ON)
 #------------------------------------------------------------------------------
 
-#Find packages needed for poll2
+#Find packages needed for the
```
for the ... ?

spaulaus replied:
shut up? I'll fix it.

ksmith0 commented on CMakeLists.txt:
```diff
@@ -112,8 +115,7 @@ if (BUILD_SUITE)
 		#Find the PLX Library
 	   find_package(PLX REQUIRED)
-   	link_directories(${PLX_LIBRARY_DIR})
-	   #add_definitions("-DPLX_LINUX -DPCI_CODE -DPLX_LITTLE_ENDIAN")
```
This line was here as carried over from PixieSuite1. Do we know what it did and that it was safe to remove them?

spaulaus replied:
It was commented out. I thought I just removed the comment. I will add back in the link_directory line.

ksmith0 replied:
Yes it was commented out because we were not sure of its function. I don't want to throw it away unless we know we don't need it.

spaulaus replied:
If we've gotten this far without it, then I suspect that we do not need it. I would prefer not to have blocks of code commented out.

ksmith0 commented on CMakeLists.txt:
```diff
@@ -141,38 +143,30 @@ if (USE_ROOT)
-	add_subdirectory(Poll)
+#Build Acquisition software, we don't have this broken into as fine of a
+#granularity as the Analysis software, so we just wrap the whole thing in an if.
+if (BUILD_SUITE)
```
Should we change BUILD_SUITE to BUILD_ACQ?

spaulaus replied:
You know, I had started to do that, but changed my mind midway through. I stopped because I didn't want to figure out what/how to change things for the Scan part.  **I'll change it. It makes sense.**

ksmith0 commented on ThirdParty/source/CMakeLists.txt:
```diff
@@ -0,0 +1,9 @@
+set(ThirdPartySources pugixml.cpp)
+
+add_library(ThirdPartyObjects OBJECT ${ThirdPartySources})
+add_library(ThirdPartyStatic STATIC $<TARGET_OBJECTS:ThirdPartyObjects>)
```
I'm not sure these names make sense. I doubt you will often want to link against all the third party things at once. One library for each makes more sense to me.

spaulaus replied:
No problem. I'll make the change.

spaulaus committed:
Changing how 3rd party libraries are built. Fixing main CMakeLists

The CMakeLists.txt had a few things that needed tweaked for the code review. I renamed BUILD_SUITE to BUILD_ACQ. This now removes nearly all of the the PixieSuite left overs. I also cleaned up some of the comments. 

The ThirdParty libraries are now built into individual libraries instead of one big one. This should make it easier for users who only need specific libraries to link against them.

spaulaus committed:
Forgot to stage a change to the utkscan CMakeLists

spaulaus commented:
@ksmith0 : I have addressed each of your comments above. I have not added back in the commented line since I'm not convinced that we need it.

ksmith0 reviewed (approved):
Compiled with cmake 2 and 3 on r230 and takk. With and without HRIBF.
